### ISSUE TITLE
Support for ice, iavf

### ISSUE THREAD

**Issue by jbrandeb** [isc-projects/ethq#27]
Please consider the patch in #26 to solve this issue:

Driver info (5.19 Fedora kernel)
``` 
# ethtool -i enp59s0f0
driver: ice
version: 5.19.4-100.fc35.x86_64
firmware-version: 4.22 0x80010446 1.3197.0
expansion-rom-version:
bus-info: 0000:3b:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: yes

# ethtool -i enp59s0f0v0
driver: iavf
version: 5.19.4-100.fc35.x86_64
firmware-version: N/A
expansion-rom-version:
bus-info: 0000:3b:01.0
supports-statistics: yes
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: yes
```

**ethtool -S**
ice:
```
NIC statistics:
     rx_unicast: 0
     tx_unicast: 0
     rx_multicast: 0
     tx_multicast: 18
     rx_broadcast: 40
     tx_broadcast: 3
     rx_bytes: 13320
     tx_bytes: 1600
     rx_dropped: 0
     rx_unknown_protocol: 0
     rx_alloc_fail: 0
     rx_pg_alloc_fail: 0
     tx_errors: 0
     tx_linearize: 0
     tx_busy: 0
     tx_restart: 0
     tx_queue_0_packets: 0
     tx_queue_0_bytes: 0
     tx_queue_1_packets: 0
     tx_queue_1_bytes: 0
     tx_queue_2_packets: 1
     tx_queue_2_bytes: 62
     tx_queue_3_packets: 0
     tx_queue_3_bytes: 0
     tx_queue_4_packets: 0
     tx_queue_4_bytes: 0
     tx_queue_5_packets: 0
     tx_queue_5_bytes: 0
     tx_queue_6_packets: 0
     tx_queue_6_bytes: 0
     tx_queue_7_packets: 0
     tx_queue_7_bytes: 0
     tx_queue_8_packets: 0
     tx_queue_8_bytes: 0
     tx_queue_9_packets: 0
     tx_queue_9_bytes: 0
     tx_queue_10_packets: 0
     tx_queue_10_bytes: 0
     tx_queue_11_packets: 0
     tx_queue_11_bytes: 0
     tx_queue_12_packets: 0
     tx_queue_12_bytes: 0
     tx_queue_13_packets: 0
     tx_queue_13_bytes: 0
     tx_queue_14_packets: 0
     tx_queue_14_bytes: 0
     tx_queue_15_packets: 0
     tx_queue_15_bytes: 0
     tx_queue_16_packets: 0
     tx_queue_16_bytes: 0
     tx_queue_17_packets: 0
     tx_queue_17_bytes: 0
     tx_queue_18_packets: 0
     tx_queue_18_bytes: 0
     tx_queue_19_packets: 0
     tx_queue_19_bytes: 0
     tx_queue_20_packets: 0
     tx_queue_20_bytes: 0
     tx_queue_21_packets: 0
     tx_queue_21_bytes: 0
     tx_queue_22_packets: 0
     tx_queue_22_bytes: 0
     tx_queue_23_packets: 0
     tx_queue_23_bytes: 0
     tx_queue_24_packets: 0
     tx_queue_24_bytes: 0
     tx_queue_25_packets: 0
     tx_queue_25_bytes: 0
     tx_queue_26_packets: 0
     tx_queue_26_bytes: 0
     tx_queue_27_packets: 0
     tx_queue_27_bytes: 0
     tx_queue_28_packets: 2
     tx_queue_28_bytes: 124
     tx_queue_29_packets: 1
     tx_queue_29_bytes: 62
     tx_queue_30_packets: 5
     tx_queue_30_bytes: 402
     tx_queue_31_packets: 0
     tx_queue_31_bytes: 0
     tx_queue_32_packets: 0
     tx_queue_32_bytes: 0
     tx_queue_33_packets: 0
     tx_queue_33_bytes: 0
     tx_queue_34_packets: 0
     tx_queue_34_bytes: 0
     tx_queue_35_packets: 0
     tx_queue_35_bytes: 0
     tx_queue_36_packets: 0
     tx_queue_36_bytes: 0
     tx_queue_37_packets: 0
     tx_queue_37_bytes: 0
     tx_queue_38_packets: 0
     tx_queue_38_bytes: 0
     tx_queue_39_packets: 0
     tx_queue_39_bytes: 0
     tx_queue_40_packets: 0
     tx_queue_40_bytes: 0
     tx_queue_41_packets: 0
     tx_queue_41_bytes: 0
     tx_queue_42_packets: 1
     tx_queue_42_bytes: 62
     tx_queue_43_packets: 3
     tx_queue_43_bytes: 182
     tx_queue_44_packets: 0
     tx_queue_44_bytes: 0
     tx_queue_45_packets: 0
     tx_queue_45_bytes: 0
     tx_queue_46_packets: 0
     tx_queue_46_bytes: 0
     tx_queue_47_packets: 0
     tx_queue_47_bytes: 0
     tx_queue_48_packets: 0
     tx_queue_48_bytes: 0
     tx_queue_49_packets: 0
     tx_queue_49_bytes: 0
     tx_queue_50_packets: 8
     tx_queue_50_bytes: 706
     tx_queue_51_packets: 0
     tx_queue_51_bytes: 0
     tx_queue_52_packets: 0
     tx_queue_52_bytes: 0
     tx_queue_53_packets: 0
     tx_queue_53_bytes: 0
     tx_queue_54_packets: 0
     tx_queue_54_bytes: 0
     tx_queue_55_packets: 0
     tx_queue_55_bytes: 0
     rx_queue_0_packets: 0
     rx_queue_0_bytes: 0
     rx_queue_1_packets: 0
     rx_queue_1_bytes: 0
     rx_queue_2_packets: 0
     rx_queue_2_bytes: 0
     rx_queue_3_packets: 0
     rx_queue_3_bytes: 0
     rx_queue_4_packets: 0
     rx_queue_4_bytes: 0
     rx_queue_5_packets: 0
     rx_queue_5_bytes: 0
     rx_queue_6_packets: 0
     rx_queue_6_bytes: 0
     rx_queue_7_packets: 0
     rx_queue_7_bytes: 0
     rx_queue_8_packets: 0
     rx_queue_8_bytes: 0
     rx_queue_9_packets: 40
     rx_queue_9_bytes: 13320
     rx_queue_10_packets: 0
     rx_queue_10_bytes: 0
     rx_queue_11_packets: 0
     rx_queue_11_bytes: 0
     rx_queue_12_packets: 0
     rx_queue_12_bytes: 0
     rx_queue_13_packets: 0
     rx_queue_13_bytes: 0
     rx_queue_14_packets: 0
     rx_queue_14_bytes: 0
     rx_queue_15_packets: 0
     rx_queue_15_bytes: 0
     rx_queue_16_packets: 0
     rx_queue_16_bytes: 0
     rx_queue_17_packets: 0
     rx_queue_17_bytes: 0
     rx_queue_18_packets: 0
     rx_queue_18_bytes: 0
     rx_queue_19_packets: 0
     rx_queue_19_bytes: 0
     rx_queue_20_packets: 0
     rx_queue_20_bytes: 0
     rx_queue_21_packets: 0
     rx_queue_21_bytes: 0
     rx_queue_22_packets: 0
     rx_queue_22_bytes: 0
     rx_queue_23_packets: 0
     rx_queue_23_bytes: 0
     rx_queue_24_packets: 0
     rx_queue_24_bytes: 0
     rx_queue_25_packets: 0
     rx_queue_25_bytes: 0
     rx_queue_26_packets: 0
     rx_queue_26_bytes: 0
     rx_queue_27_packets: 0
     rx_queue_27_bytes: 0
     rx_queue_28_packets: 0
     rx_queue_28_bytes: 0
     rx_queue_29_packets: 0
     rx_queue_29_bytes: 0
     rx_queue_30_packets: 0
     rx_queue_30_bytes: 0
     rx_queue_31_packets: 0
     rx_queue_31_bytes: 0
     rx_queue_32_packets: 0
     rx_queue_32_bytes: 0
     rx_queue_33_packets: 0
     rx_queue_33_bytes: 0
     rx_queue_34_packets: 0
     rx_queue_34_bytes: 0
     rx_queue_35_packets: 0
     rx_queue_35_bytes: 0
     rx_queue_36_packets: 0
     rx_queue_36_bytes: 0
     rx_queue_37_packets: 0
     rx_queue_37_bytes: 0
     rx_queue_38_packets: 0
     rx_queue_38_bytes: 0
     rx_queue_39_packets: 0
     rx_queue_39_bytes: 0
     rx_queue_40_packets: 0
     rx_queue_40_bytes: 0
     rx_queue_41_packets: 0
     rx_queue_41_bytes: 0
     rx_queue_42_packets: 0
     rx_queue_42_bytes: 0
     rx_queue_43_packets: 0
     rx_queue_43_bytes: 0
     rx_queue_44_packets: 0
     rx_queue_44_bytes: 0
     rx_queue_45_packets: 0
     rx_queue_45_bytes: 0
     rx_queue_46_packets: 0
     rx_queue_46_bytes: 0
     rx_queue_47_packets: 0
     rx_queue_47_bytes: 0
     rx_queue_48_packets: 0
     rx_queue_48_bytes: 0
     rx_queue_49_packets: 0
     rx_queue_49_bytes: 0
     rx_queue_50_packets: 0
     rx_queue_50_bytes: 0
     rx_queue_51_packets: 0
     rx_queue_51_bytes: 0
     rx_queue_52_packets: 0
     rx_queue_52_bytes: 0
     rx_queue_53_packets: 0
     rx_queue_53_bytes: 0
     rx_queue_54_packets: 0
     rx_queue_54_bytes: 0
     rx_queue_55_packets: 0
     rx_queue_55_bytes: 0
     rx_bytes.nic: 0
     tx_bytes.nic: 24528
     rx_unicast.nic: 0
     tx_unicast.nic: 0
     rx_multicast.nic: 0
     tx_multicast.nic: 116
     rx_broadcast.nic: 0
     tx_broadcast.nic: 43
     tx_errors.nic: 0
     tx_timeout.nic: 0
     rx_size_64.nic: 0
     tx_size_64.nic: 5
     rx_size_127.nic: 0
     tx_size_127.nic: 106
     rx_size_255.nic: 0
     tx_size_255.nic: 8
     rx_size_511.nic: 0
     tx_size_511.nic: 40
     rx_size_1023.nic: 0
     tx_size_1023.nic: 0
     rx_size_1522.nic: 0
     tx_size_1522.nic: 0
     rx_size_big.nic: 0
     tx_size_big.nic: 0
     link_xon_rx.nic: 0
     link_xon_tx.nic: 0
     link_xoff_rx.nic: 0
     link_xoff_tx.nic: 0
     tx_dropped_link_down.nic: 0
     rx_undersize.nic: 0
     rx_fragments.nic: 0
     rx_oversize.nic: 0
     rx_jabber.nic: 0
     rx_csum_bad.nic: 0
     rx_length_errors.nic: 0
     rx_dropped.nic: 0
     rx_crc_errors.nic: 0
     illegal_bytes.nic: 0
     mac_local_faults.nic: 0
     mac_remote_faults.nic: 0
     fdir_sb_match.nic: 0
     fdir_sb_status.nic: 1
     tx_priority_0_xon.nic: 0
     tx_priority_0_xoff.nic: 0
     tx_priority_1_xon.nic: 0
     tx_priority_1_xoff.nic: 0
     tx_priority_2_xon.nic: 0
     tx_priority_2_xoff.nic: 0
     tx_priority_3_xon.nic: 0
     tx_priority_3_xoff.nic: 0
     tx_priority_4_xon.nic: 0
     tx_priority_4_xoff.nic: 0
     tx_priority_5_xon.nic: 0
     tx_priority_5_xoff.nic: 0
     tx_priority_6_xon.nic: 0
     tx_priority_6_xoff.nic: 0
     tx_priority_7_xon.nic: 0
     tx_priority_7_xoff.nic: 0
     rx_priority_0_xon.nic: 0
     rx_priority_0_xoff.nic: 0
     rx_priority_1_xon.nic: 0
     rx_priority_1_xoff.nic: 0
     rx_priority_2_xon.nic: 0
     rx_priority_2_xoff.nic: 0
     rx_priority_3_xon.nic: 0
     rx_priority_3_xoff.nic: 0
     rx_priority_4_xon.nic: 0
     rx_priority_4_xoff.nic: 0
     rx_priority_5_xon.nic: 0
     rx_priority_5_xoff.nic: 0
     rx_priority_6_xon.nic: 0
     rx_priority_6_xoff.nic: 0
     rx_priority_7_xon.nic: 0
     rx_priority_7_xoff.nic: 0
```

iavf:
```
NIC statistics:
     rx_bytes: 5994
     rx_unicast: 0
     rx_multicast: 0
     rx_broadcast: 18
     rx_discards: 0
     rx_unknown_protocol: 0
     tx_bytes: 9732
     tx_unicast: 0
     tx_multicast: 41
     tx_broadcast: 18
     tx_discards: 0
     tx_errors: 0
     tx-0.packets: 20
     tx-0.bytes: 6660
     rx-0.packets: 0
     rx-0.bytes: 0
     tx-1.packets: 16
     tx-1.bytes: 1680
     rx-1.packets: 0
     rx-1.bytes: 0
     tx-2.packets: 0
     tx-2.bytes: 0
     rx-2.packets: 0
     rx-2.bytes: 0
     tx-3.packets: 0
     tx-3.bytes: 0
     rx-3.packets: 0
     rx-3.bytes: 0
     tx-4.packets: 0
     tx-4.bytes: 0
     rx-4.packets: 0
     rx-4.bytes: 0
     tx-5.packets: 0
     tx-5.bytes: 0
     rx-5.packets: 0
     rx-5.bytes: 0
     tx-6.packets: 16
     tx-6.bytes: 992
     rx-6.packets: 0
     rx-6.bytes: 0
     tx-7.packets: 4
     tx-7.bytes: 344
     rx-7.packets: 0
     rx-7.bytes: 0
     tx-8.packets: 0
     tx-8.bytes: 0
     rx-8.packets: 0
     rx-8.bytes: 0
     tx-9.packets: 0
     tx-9.bytes: 0
     rx-9.packets: 19
     rx-9.bytes: 6327
     tx-10.packets: 0
     tx-10.bytes: 0
     rx-10.packets: 0
     rx-10.bytes: 0
     tx-11.packets: 0
     tx-11.bytes: 0
     rx-11.packets: 0
     rx-11.bytes: 0
     tx-12.packets: 0
     tx-12.bytes: 0
     rx-12.packets: 0
     rx-12.bytes: 0
     tx-13.packets: 0
     tx-13.bytes: 0
     rx-13.packets: 0
     rx-13.bytes: 0
     tx-14.packets: 0
     tx-14.bytes: 0
     rx-14.packets: 0
     rx-14.bytes: 0
     tx-15.packets: 13
     tx-15.bytes: 1470
     rx-15.packets: 0
     rx-15.bytes: 0
```

**Comment by jbrandeb** [comment]
@raybellis is there a chance you can look at this and the patch I provided?

**Comment by raybellis** [comment]
Jesse, first of all, thanks for the patch and I'm really glad to see Intel taking notice of this.

I have two requests relating to the patch:

- please also add the `ethtool -S` output files above into the `tests/` directory
- ideally please also remove the (C) Intel - the additions to this file are just data (most of which is copied from other device definitions) and aren't in our opinion copyrightable material.

If the latter causes you problems with your own legal, please get back to me direct (ray@isc.org) so we can discuss a resolution.

Thanks!

Ray

**Comment by raybellis** [comment]
@jbrandeb ?

**Comment by jbrandeb** [comment]
> * please also add the `ethtool -S` output file for `iavf` into the `tests/` directory

done, for both ice and iavf. I also tried running the tests and I think they work.

> * ideally please also remove the (C) Intel - the additions to this file are just data (most of which is copied from other device definitions) and aren't in our opinion copyrightable material.

done

I've pushed an update to my series, now that I know about the tests, I've double checked those work, and improved the regular expressions for ice and iavf, so that the totals parsing all works. 

The intent is that you squash merge the series from my fork, if you're ok to pick up the gitignore patch as part of that.

**Comment by raybellis** [comment]
Thanks Jesse - acking receipt, will look at this further when I return to work next week.

**Comment by raybellis** [comment]
Closed by 8b0d84f88d77628e12ff85e14a9ba7d71aa123d4

Thanks for the contribution, Jesse!
